src/modules/solution-explorer/solution-explorer-solution/solution-explorer-solution.ts

```diff
@@ -250,6 +253,47 @@ export class SolutionExplorerSolution {
     this._renameDiagramInput = input;
   }
 
+  public activateContextMenu(event: MouseEvent, diagram: IDiagram): void {
+    this._diagramInContextMenu = diagram;
+
+    this.diagramContextMenu.style.top = `${event.y}px`;
+    this.diagramContextMenu.style.left = `${event.x}px`;
+    this.showContextMenu = true;
+
+    const documentEventListener: EventListenerOrEventListenerObject = (): void => {
+      this.showContextMenu = false;
+      this._diagramInContextMenu = undefined;
+
+      document.removeEventListener('click', documentEventListener);
+    };
+
+    document.addEventListener('click', documentEventListener);
+  }
+
+  public async copyDiagram(): Promise<void> {
+    if (!this._diagramInContextMenu) { return; }
```

Comment:
```suggestion
    if (!this._diagramInContextMenu) {
    
    return;
  }
```

Also, please extract the evaluation rule here.